README.md

```diff
@@ -246,6 +246,70 @@ client.xml("denominator", "secret"); // <login "user_name"="denominator" "passwo
 client.json("denominator", "secret"); // {"user_name": "denominator", "password": "secret"}
 ```
 
+### Headers
+Feign supports settings headers on requests either as part of the api or as part of the client
+dending on the use case.
+
+#### Set headers using apis
+In cases where specific interfaces or calls should always have certain header values set, it
+makes sense to define headers as part of the api.
+
+Static headers can be set on an api interface or method using the `@Headers` annotation.
+
+```java
+@Headers("Accept: application/json")
+interface BaseApi<V> {
+  @Headers("Content-Type: application/json")
+  @RequestLine("PUT /api/{key}")
+  void put(@Param("key") String, V value);
+}
+```
+
+Methods can specify dynamic content for static headers using using variable expansion in `@Headers`.
+
+```java
+ @RequestLine("POST /")
+ @Headers({
+  "X-Foo: Bar",
+  "X-Ping: {token}"
+ }) void post(@Param("token") String token);
+```
+
+These approaches specify specific header entries as part of the api without requiring any cutomizations
+when buildling Feing clients. It is not currently possible to customize the header entries themselves
+on a per-request basis at the api level.
+
+#### Set headers using client
+In cases where headers should differ for the same api based on different endpoints or where per-request
+customization is required, headers can be set as part of the client using a `RequestInterceptor` or a
+`Target`.
+
+For an example of setting headers using a `RequestInterceptor`, see the `Request Interceptors` section.
+
+Headers can be set as part of a custom `Target`.
+
+```java
+  static class ForwardedForTarget extends HardCodedTarget {
+    public ForwardedForTarget(Class type, String url) {
+      super(type, url);
+    }
+
+    @Override
+    public Request apply(RequestTemplate input) {
+      input.header("X-Forwarded-For", "origin.host.com");
```

Comment:
not a very realistic target..

Target's main use has been adding authentication headers or anything tha varies on the url.
Have a look at denominator for examples. https://github.com/Netflix/denominator/find/master